File: server/routes/sms.ts
import { Router, Request, Response } from 'express';
import { sendVerificationCode, verifyCode, updateSmsConfig, getSmsConfig } from '../services/aliyunSms';

const router = Router();

// 发送验证码
router.post('/send', async (req: Request, res: Response) => {
    const { phone } = req.body;

    // 验证手机号格式
    if (!phone || !/^1[3-9]\d{9}$/.test(phone)) {
        return res.status(400).json({
            success: false,
            message: '请输入有效的 11 位手机号'
        });
    }

    try {
        const result = await sendVerificationCode(phone);
        res.json(result);
    } catch (error: any) {
        console.error('Send SMS Error:', error);
        res.status(500).json({
            success: false,
            message: '服务器错误，请稍后再试'
        });
    }
});

// 验证验证码
router.post('/verify', (req: Request, res: Response) => {
    const { phone, code } = req.body;

    if (!phone || !code) {
        return res.status(400).json({
            success: false,
            message: '手机号和验证码不能为空'
        });
    }

    const result = verifyCode(phone, code);
    res.json(result);
});

// 获取短信配置 (后台管理用)
router.get('/config', (req: Request, res: Response) => {
    const config = getSmsConfig();
    // 不返回完整密钥，只返回部分用于显示
    res.json({
        success: true,
        config: {
            accessKeyId: config.accessKeyId ? config.accessKeyId.slice(0, 6) + '****' : '',
            accessKeySecret: config.accessKeySecret ? '********' : '',
            signName: config.signName || '',
            templateCode: config.templateCode || '',
            isConfigured: !!(config.accessKeyId && config.accessKeySecret)
        }
    });
});

// 更新短信配置 (后台管理用)
router.post('/config', (req: Request, res: Response) => {
    const { accessKeyId, accessKeySecret, signName, templateCode } = req.body;

    if (!accessKeyId || !accessKeySecret || !signName || !templateCode) {
        return res.status(400).json({
            success: false,
            message: '所有字段都是必填的'
        });
    }

    try {
        updateSmsConfig({ accessKeyId, accessKeySecret, signName, templateCode });
        res.json({ success: true, message: '配置已保存' });
    } catch (error: any) {
        console.error('Save Config Error:', error);
        res.status(500).json({
            success: false,
            message: '保存失败: ' + error.message
        });
    }
});

export default router;

File: server_py/routers/sms.py
from fastapi import APIRouter, Depends, HTTPException
from sqlmodel import Session, select
from typing import List, Optional
from ..db import get_session
from ..models import Setting, User
from ..services.sms_service import SMSService
from .auth import get_current_admin
import json
import os

router = APIRouter()

@router.get("/config")
async def get_sms_config(session: Session = Depends(get_session)):
    """前端获取SMS配置状态"""
    setting = session.get(Setting, "sms_config")
    if not setting:
        return {
            "success": True,
            "config": {
                "accessKeyId": "",
                "signName": "",
                "templateCode": "",
                "isConfigured": False
            }
        }
    try:
        data = json.loads(setting.value)
        return {
            "success": True,
            "config": {
                "accessKeyId": data.get("accessKeyId", "")[:4] + "****" if data.get("accessKeyId") else "",
                "signName": data.get("signName", ""),
                "templateCode": data.get("templateCode", ""),
                "isConfigured": bool(data.get("accessKeyId") or data.get("appCode"))
            }
        }
    except:
        return {"success": False, "config": {"isConfigured": False}}

@router.post("/config")
@router.post("/config/")
async def save_sms_config(
    config: dict,
    session: Session = Depends(get_session),
    admin: User = Depends(get_current_admin)
):
    """保存SMS配置"""
    setting = session.get(Setting, "sms_config")
    if not setting:
        setting = Setting(key="sms_config", value=json.dumps(config))
    else:
        setting.value = json.dumps(config)

    session.add(setting)
    session.commit()
    return {"success": True}

@router.get("/settings")
async def get_sms_settings(
    session: Session = Depends(get_session),
    admin: User = Depends(get_current_admin)
):
    setting = session.get(Setting, "sms_config")
    if not setting:
        return {
            "accessKeyId": "",
            "accessKeySecret": "",
            "signName": "",
            "templateCode": "",
            "appCode": ""
        }
    try:
        data = json.loads(setting.value)
        return data
    except:
        return {}

@router.post("/settings")
@router.post("/settings/")
async def save_sms_settings(
    settings: dict,
    session: Session = Depends(get_session),
    admin: User = Depends(get_current_admin)
):
    setting = session.get(Setting, "sms_config")
    if not setting:
        setting = Setting(key="sms_config", value=json.dumps(settings))
    else:
        setting.value = json.dumps(settings)
    
    session.add(setting)
    session.commit()
    return {"success": True}

@router.post("/send-code")
@router.post("/send-code/")
async def send_code(
    data: dict,
    session: Session = Depends(get_session)
):
    mobile = data.get("mobile")
    if not mobile:
        raise HTTPException(status_code=400, detail="请输入手机号")
    
    # Get SMS config for appCode
    setting = session.get(Setting, "sms_config")
    if not setting:
        raise HTTPException(status_code=500, detail="短信服务未配置")
    
    config = json.loads(setting.value)
    app_code = config.get("appCode")
    
    if not app_code:
        # Fallback to accessKeyId if appCode is not specifically set (for compatibility)
        app_code = config.get("accessKeyId")
    
    if not app_code:
        raise HTTPException(status_code=500, detail="配置中缺少阿里云 AppCode")
    
    success = await SMSService.send_verification_code(mobile, session, app_code)
    if not success:
        raise HTTPException(status_code=500, detail="验证码发送失败，请检查短信服务配置。")
        
    return {"success": True, "message": "验证码已发送"}

@router.post("/send-test")
@router.post("/send-test/")
async def send_test_sms(
    data: dict,
    session: Session = Depends(get_session),
    admin: User = Depends(get_current_admin)
):
    phone = data.get("phone")
    if not phone:
        raise HTTPException(status_code=400, detail="请输入手机号")
    
    return {"success": True, "message": f"Mock test SMS sent to {phone}"}
